source/fundamentals/authentication/mechanisms.txt

```diff
@@ -235,6 +235,22 @@ The driver checks for your credentials in the following sources in order:
       .. literalinclude:: /code-snippets/authentication/aws-env-variable.js
          :language: javascript
 
+.. important:: Retrieval of AWS Credentials
+   
+   Starting in version 4.11, when you install the optional
+   ``aws-sdk/credential-providers`` dependency, the driver uses the AWS SDK
+   to retrieve credentials from the environment. As a result, if you
+   have a shared AWS credentials file or config file, the driver will
+   use those credentials by default if there are no AWS authentication
+   environment variables set.
+   
+   To override this behavior, you can perform one of the following actions:
+   
+   - Set ``AWS_SHARED_CREDENTIALS_FILE`` in your shell or set the
+     equivalent environment variable value in your application.
+   - Create an AWS profile for your MongoDB credentials and set the
+     ``AWS_PROFILE`` environment variable to that profile name.
+
```

Comment:
Question

What if you do both of these options? Which one has precedence? Might be beyond the scope or not something you can answer, but was a thought that came to mind.

Reply:
Yes, it might be out of scope. I'm not sure what the order is